### ISSUE TITLE
Finalize NFT Marketplace

### ISSUE THREAD

**Issue by green-jay** [galacticcouncil/Basilisk-node#436]
Legacy PRs:
https://github.com/galacticcouncil/Basilisk-node/pull/137 (merged for testing purposes)

- [ ] Address any pending comments in legacy PRs
- [ ] Do full review of pallet-marketplace (ideally multiple peers)
- [ ] Rebenchmark on reference machine
- [ ] Remove Marketplace call filter in basilisk runtime

**Comment by dmoka** [comment]
I am going to pick this up as discussed with @lumir-mrkva that we want to utilize auctions soon on Basilisk.

**Comment by green-jay** [comment]
> I am going to pick this up as discussed with @lumir-mrkva that we want to utilize auctions soon on Basilisk.

The status quo of the marketplace chaged since April. So the list should include additionally:
- [ ] migrate to the latest version of pallet-nft
- [ ] refactor the marketplace to support multiple currencies

**Comment by dmoka** [comment]
@green-jay , as for pallet-nft, I just updated to the latest version and everything is still functional, so no breaking changes. I guess it is fine, or did you expect some actual code changes in the marketplace pallet (because of some important behavioral change in the pallet-nft)?

**Comment by green-jay** [comment]
> @green-jay , as for pallet-nft, I just updated to the latest version and everything is still functional, so no breaking changes. I guess it is fine, or did you expect some actual code changes in the marketplace pallet (because of some important behavioral change in the pallet-nft)?

as long as it works with latest changes should be fine

**Comment by green-jay** [comment]
https://github.com/kodadot/nft-gallery/issues?q=is%3Aissue+is%3Aopen+label%3A%22%F0%9F%90%8D+-+basilisk%22
Also these issues should be resolved

**Comment by enthusiastmartin** [comment]
Completed?

**Comment by dmoka** [comment]
> Completed?

Nope, we froze the work regarding this, as it had low priority. Some work has been done on branch `feat/finalize-nft-marketplace`, but not complete. 

We are planning to add [multicurrency support](https://www.notion.so/Support-multiple-currencies-dd266dd531e44d74ab4a9648204865a9) which would require some work. The plan would be to extend the nft marketplace functionality without breaking the existing APIs.
